### PR TITLE
Refacto

### PR DESCRIPTION
**Pull request by blackout1471**

### CODE REVIEW

File: Source/MqttPackages/ConnectAckPackage.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #include "ConnectAckVariableHeader.h"
 
 namespace MQTT {
-	namespace Protocol {
+	namespace MqttPackages {
```
Comment (Ripiter) [inline Source/MqttPackages/ConnectAckPackage.h:6]: mqtt in mqttpackages redundant?

File: Source/Protocol/Mqtt311Handler.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+namespace MQTT {
+	namespace Protocol {
+		Mqtt311Handler::Mqtt311Handler() : m_ClientStates(), m_Manager(), m_SubscribeManager()
+		{
+		}
+
+		Mqtt311Handler::~Mqtt311Handler()
+		{
+			for (auto& clientState : m_ClientStates)
```
Comment (Ripiter) [inline Source/Protocol/Mqtt311Handler.cpp]: Deleting reference and not the pointer?

Comment (Ripiter) [reply Source/Protocol/Mqtt311Handler.cpp]: should vector be cleaned?

File: Source/Rules/BufferValidRule.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include "IRule.h"
+
+#include <vector>
+
+namespace MQTT {
+	namespace Rules {
```
Comment (Ripiter) [inline Source/Rules/BufferValidRule.h:7]: Shouldnt rules be under protocol?

Comment (blackout1471) [reply Source/Rules/BufferValidRule.h:7]: No according to the class diagram it should just be Rules

File: Source/Protocol/Mqtt311Handler.cpp (hunk 2 above)
Comment (blackout1471) [reply Source/Protocol/Mqtt311Handler.cpp]: done

Comment (blackout1471) [reply Source/Protocol/Mqtt311Handler.cpp]: Vector, doesn't have to be deleted as it is created on the stack, and will automatically be deleted when destructor is called.

**Review by Ripiter — APPROVED** [review]
Looks good

**Pull request merged**